File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2327.cpp
#include <bits/stdc++.h>
#define ff first
#define ss second
#define endl '\n'
#define INF 1e9

using namespace std;
using ll = long long;
using pii = pair<int, int>;
using vi = vector<int>;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    
    ll n, v[16][16];
    cin >> n;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cin >> v[i][j];
        }
    }
    
    ll linhas[16] = {}, colunas[16] = {}, diag1 = 0, diag2 = 0;
    
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            linhas[j] += v[i][j];
        }
    }
    
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            colunas[j] += v[j][i];
        }
    }
    
    for (int i = 0; i < n; i++) {
        diag1 += v[i][i];
    }
    
    for (int i = 0; i < n; i++) {
        diag2 += v[n-i-1][i];
    }
    
    if (diag1 != diag2) {
        cout << -1 << endl;
        return 0;
    }
    
    for (int i = 0; i < n; i++) {
        if (linhas[i] != diag1 or colunas[i] != diag1) {
            cout << -1 << endl;
            return 0;
        }
    }
    
    cout << diag1 << endl;
    
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2327.py
def main():
    n = int(input())
    
    v = []
    
    for i in range(n):
        v.append([int(i) for i in input().split()])
    
    linhas = []
    colunas = []
    diag1 = diag2 = 0
    
    for i in range(n):
        linhas.append(0)
        for j in range(n):
            linhas[-1] += v[i][j]
            
    for i in range(n):
        colunas.append(0)
        for j in range(n):
            colunas[-1] += v[j][i]
    
    for i in range(n):
        diag1 += v[i][i]
        
    for i in range(n):
        diag2 += v[n-1-i][i]
        
    if diag1 != diag2:
        print(-1)
        return
    
    for i in range(n):
        if linhas[i] != diag1 or colunas[i] != diag1:
            print(-1)
            return
    
    print(diag1)
    
main()